text/0000-destructure-drop-structs.md

```diff
@@ -0,0 +1,58 @@
+- Feature Name: Destructure structs implementing Drop
+- Start Date: 2017-07-08
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow destructuring of structs that implement Drop.
+
+# Motivation
+[motivation]: #motivation
+
+Currently destructuring of structs that implement Drop, requires to read non-copy fields via 
+`ptr::read(&struct.field)` followed by `mem::forget(struct)`.
+
+This leaves more room for error, than there would have to be: 
+1.  forgetting to read all fields, possibly creating a memory leak,
+2.  acidentally working with `&struct` instead of `struct`, leading to a double free.
+     (The fields are copied, but mem::forget only forgets the reference.)
+
+Allowing to destructure these types would ensure that:
+1.  unused fields are dropped,
+2.  only owned structs can be destructured.
+
+# Detailed design
+[design]: #detailed-design
```

Comment:
The level of detail here seems low. Which solution is this RFC proposing? What are the pros and cons of the various options?